src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java

```diff
@@ -85,88 +89,169 @@ public int getLinesPerPage(MessageReceiver source) {
-            int j = UNICODE_CHAR_WIDTHS[codePoint] >>> 4;
-            int k = UNICODE_CHAR_WIDTHS[codePoint] & 15;
+        } else if (UNICODE_CHAR_WIDTHS[codePoint] != 0) {
+            //from 1.9 & 255 to avoid strange signed int math ruining things.
+            //https://bugs.mojang.com/browse/MC-7181
+            final int temp = UNICODE_CHAR_WIDTHS[codePoint] & 255;
+            // Split into high and low nibbles.
+            //bit digits
+            //87654321 >>> 4 = 00008765
+            int startColumn = temp >>> 4;
+            //87654321 & 00001111 = 00004321
+            int endColumn = temp & 15;
 
-            if (k > 7) {
-                k = 15;
-                j = 0;
-            }
-            width = ((k + 1) - j) / 2 + 1;
-            if (isBold) {
-                width += 0.5;
-            }
+            width = (endColumn + 1) - startColumn;
+            //Why does this scaling happen?
+            //I believe it makes unicode fonts skinnier to better match the character widths of the default Minecraft
+            // font however there is a int math vs float math bug in the Minecraft FontRenderer.
+            //The float math is adjusted for rendering, they attempt to do the same thing for calculating string widths
+            //using integer math, this has potential rounding errors, but we should copy it and use ints as well.
+            width = (width / 2) + 1;
+        } else {
+            width = 0;
         }
+        //if bolded width gets 1 added.
+        if(isBold && width > 0) ++width;
+
         return width;
     }
 
-    private int getLength(MessageReceiver source, Text text) {
-        double columnCount = 0d;
-        for (Text child : text.withChildren()) {
-            final String txt;
+    @VisibleForTesting
+    int getLength(Text text) {
+        Iterator<Text> it = text.withChildren().iterator();
+        int total = 0;
+        String debug = "";
+        while(it.hasNext()){
+            Text child = it.next();
+            String str;
+            PrimitiveIterator.OfInt i_it;
             if (child instanceof LiteralText) {
-                txt = ((LiteralText) child).getContent();
+                i_it = ((LiteralText) child).getContent().codePoints().iterator();
+                str = ((LiteralText) child).getContent();
             } else if (child instanceof TranslatableText) {
-                txt = child.toPlain();
+                i_it = child.toPlain().codePoints().iterator();
+                str = child.toPlain();
             } else {
                 continue;
             }
-            boolean isBold = child.getStyle().contains(TextStyles.BOLD);
-            for (int i = 0; i < txt.length(); ++i) {
-                columnCount += getWidth(txt.codePointAt(i), isBold);
+
+            IChatComponent childComponents = SpongeTexts.toComponent(child);
+            Boolean bold = childComponents
+                    .getChatStyle()
+                    .bold;
```

Comment:
You need to use `.getBold()` or however that is called to get the inherited style